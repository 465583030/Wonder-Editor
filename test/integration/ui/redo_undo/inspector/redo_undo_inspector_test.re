open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe(
    "redo_undo: inspector",
    () => {
      let sandbox = getSandboxDefaultVal();
      let _buildMainEditorInspector = () =>
        ReactTestRenderer.create(
          <MainEditorInspector
            store=(TestToolUI.buildEmptyAppState())
            dispatch=(TestToolUI.getDispatch())
            allShowComponentsConfig=(InspectorToolUI.buildFakeGameObjectComponentRecord())
          />
        );
      beforeEach(
        () => {
          sandbox := createSandbox();
          TestToolEngine.prepare(sandbox)
        }
      );
      afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
      describe(
        "test simulate set currentGameObject",
        () => {
          let _buildEngineSceneTree = () =>
            ReactTestRenderer.create(
              <MainEditorSceneTree
                store=(SceneTreeToolUI.buildAppStateSceneGraphFromEngine())
                dispatch=(TestToolUI.getDispatch())
              />
            );
          let _setSpecificGameObject = (clickTreeNodeIndex) => {
            let component = _buildEngineSceneTree();
            EventToolUI.triggerComponentEvent(
              component,
              SceneTreeEventTool.triggerClickEvent(clickTreeNodeIndex)
            )
          };
          describe(
            "test snapshot",
            () => {
              beforeEach(
                () => {
                  TestToolEditor.closeContractCheck();
                  TestToolUI.initMainEditor(sandbox);
                  MainEditorSceneToolEditor.prepareDefaultScene(() => ());
                  StateHistoryToolEditor.clearAllState();
                  _setSpecificGameObject(1)
                }
              );
              afterEach(() => TestToolEditor.openContractCheck());
              describe(
                "test undo operate",
                () => {
                  test(
                    "test not undo",
                    () => {
                      let component = _buildMainEditorInspector();
                      let json = ReactTestRenderer.toJSON(component);
                      toMatchSnapshot(expect(json))
                    }
                  );
                  describe(
                    "test undo one step",
                    () =>
                      test(
                        "step from second to first",
                        () => {
                          StateHistoryToolEditor.undo();
                          let component = _buildMainEditorInspector();
                          let json = ReactTestRenderer.toJSON(component);
                          toMatchSnapshot(expect(json))
                        }
                      )
                  );
                  describe(
                    "test undo two step",
                    () =>
                      test(
                        "step from second to zero",
                        () => {
                          StateHistoryToolEditor.undo();
                          StateHistoryToolEditor.undo();
                          let component = _buildMainEditorInspector();
                          let json = ReactTestRenderer.toJSON(component);
                          toMatchSnapshot(expect(json))
                        }
                      )
                  )
                }
              )
            }
          )
        }
      )
    }
  );