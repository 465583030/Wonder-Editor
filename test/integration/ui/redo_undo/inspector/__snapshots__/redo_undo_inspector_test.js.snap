// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: inspector test simulate set currentGameObject test snapshot test addComponent test undo operate test not undo 1`] = `
<article
  className="inspector-component"
>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        transform
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
    </article>
  </article>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        material
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          color : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="#ffffff"
        />
      </article>
    </article>
  </article>
  <article
    className="addable-component"
  >
    <button
      disabled={false}
      onClick={[Function]}
    >
      add component
    </button>
  </article>
</article>
`;

exports[`redo_undo: inspector test simulate set currentGameObject test snapshot test addComponent test undo operate test undo one step undo step from added to not add 1`] = `
<article
  className="inspector-component"
>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        transform
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
    </article>
  </article>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        material
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          color : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="#ffffff"
        />
      </article>
    </article>
  </article>
  <article
    className="addable-component"
  >
    <button
      disabled={false}
      onClick={[Function]}
    >
      add component
    </button>
  </article>
</article>
`;

exports[`redo_undo: inspector test simulate set currentGameObject test snapshot test undo operate test not undo 1`] = `
<article
  className="inspector-component"
>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        transform
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
    </article>
  </article>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        material
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          color : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="#ffffff"
        />
      </article>
    </article>
  </article>
  <article
    className="addable-component"
  >
    <button
      disabled={false}
      onClick={[Function]}
    >
      add component
    </button>
  </article>
</article>
`;

exports[`redo_undo: inspector test simulate set currentGameObject test snapshot test undo operate test undo one step step from second to first 1`] = `
<article
  className="inspector-component"
/>
`;

exports[`redo_undo: inspector test simulate set currentGameObject test snapshot test undo operate test undo two step step from second to zero 1`] = `
<article
  className="inspector-component"
/>
`;
